### ISSUE TITLE
Can't set an iconview model to nil - segv in IconView.SetModel

### ISSUE THREAD

**Issue by patknight** [gotk3/gotk3#358]
I want to set an IconView model to nil, add a lot of widgets to the model, and put the model back into the IconView. This saves generating updates to the IconView hundreds of times, once for each widget.

I tried passing nil to IconView.SetModel and it segv'ed in toTreeModel, even though toTreeModel is trying to check for nil. I patched IconView.SetModel to check for nil before calling toTreeModel, and this workaround was effective.

The new version of SetModel looks like:
```go
// SetModel is a wrapper around gtk_icon_view_set_model().
func (v *IconView) SetModel(model ITreeModel) {
	var mptr *C.GtkTreeModel
	if model == nil {
		mptr = nil
	} else {
		mptr = model.toTreeModel()
	}
	//C.gtk_icon_view_set_model(v.native(), model.toTreeModel())
	C.gtk_icon_view_set_model(v.native(), mptr)
}
```

Have I missed a better fix to toTreeModel itself? If not and you agree with the approach, should I fix all the calls to toTreeModel implementations before submitting a pull request? Note that IconView is the only it's convenient for me to test, but I could develop a test app to try out the various views.

**Comment by Bios-Marcel** [comment]
The C side should be able to handle `null`, according to [the documentation](https://developer.gnome.org/gtk3/stable/GtkIconView.html#gtk-icon-view-set-model) it is a valid value and will cause unsetting the model. What error exactly does it throw for you and at which point of the code? Sorry if I am a bit confused here, maybe I am missunderstanding something.

**Comment by patknight** [comment]
The C side can handle null. With my workaround, SetModel works as expected.

If anyone has examples of using SetModel with nil successfully, I'll be happy to change my app to work the way the their code does. I've only been coding in Go for a few weeks and am happy to learn the right way to do things.

On thinking about how to reply to you, I see now where the bug is.

The current code in gotk3 is
```go
// SetModel is a wrapper around gtk_icon_view_set_model().
func (v *IconView) SetModel(model ITreeModel) {
	C.gtk_icon_view_set_model(v.native(), model.toTreeModel())
}
```

When called from my app if `model` is an instance of gtk.ListStore, everything works. If `model` is nil, I think the call `model.toTreeModel()` tries to dereference the `nil` to find the correct toTreeModel implementation (I haven't looked at the generated code.) It then panics at that point before entering the C code.

My fix (in the original report) tests for nil before calling `model.toTreeModel()` and just passes the nil onto the C code.

I'm not at home now (this is a hobby project.) I'll check more tonight, but I think my workaround (coded more nicely) will be the correct fix. If you like, I can find all the SetModel implementations and fix them all?

BTW - I also forgot in the original report: go 1.12, Ubuntu 18.04, GTK current as for 18.04, gotk3 head of main.

**Comment by Bios-Marcel** [comment]
Oh, that's why I was confused, because I did take a look at the Go code, where I found this:

```go
func (v *TreeModel) toTreeModel() *C.GtkTreeModel {
    if v == nil {
        return nil
    }
    return v.native()
}
```

But yes, you seem to be right. I don't see a different way of fixing this.

**Comment by patknight** [comment]
I just thought of a slightly different way. If `v` in `toTreeModel()` was an explicit argument, then the test for nil is in one(?) place instead of in all the callers i.e.
```go
func toTreeModel(v *TreeModel) *C.GtkTreeModel {
    if v == nil {
        return nil
    }
    return v.native()
}
```
I'll need to check if that works as there are multiple toTreeModel implementations in the code, but it seems cleaner. What do you think?

**Comment by Bios-Marcel** [comment]
Closed due to merge of #359
